### ISSUE TITLE
Port run-length-encoding exercise from JavaScript track

### ISSUE THREAD

**Issue by rchavarria** [exercism/ecmascript#334]
As part of the job to make the track compatible with *Nextercism*, we identified there are missing exercises in this track that are present in the [JavaScript](https://github.com/exercism/javascript) track. One of them is [run-length-encoding](https://github.com/exercism/javascript/tree/master/exercises/run-length-encoding).

The task would be to add the `run-length-encoding` exercise to this track, copying files from the JavaScript track and adapting the code to the ECMAScript 6 standard.

Basic steps would include:

1. Copy all files from `<javascript repo>/exercises/run-length-encoding` (https://github.com/exercism/javascript) to `<ecmascript repo>/exercises/run-length-encoding` (https://github.com/exercism/ecmascript)
2. Copy the main `package.json` file (it's in the root folder of the `ecmascript` repo https://github.com/exercism/ecmascript/blob/master/package.json) to the `<ecmascript repo>/exercises/run-length-encoding` folder
3. Adapt the code in `run-length-encoding.spec.js` file to use ECMAScript 6 features (`import` modules, arrow functions, `const` and/or `let`, any other feature you think will make the code better). Additionally, the JavaScript track uses Jasmine where the ECMAScript track uses Jest.  So, `it` and `xit` must be *translated* to `test` and `xtest`. Feel free to adapt `example.js` as well, just for fun (as long as tests pass).
4. In the `README.md` of the exercise, replace sections from *Setup* and below (Setup, Making the test suite pass, Source and Submitting incomplete solutions) with similar sections from any ECMAScript exercise, for example [`accumulate` README](https://github.com/exercism/ecmascript/blob/master/exercises/accumulate/README.md). They'll include Setup, Requirements, Making the test suite pass, Source and Submitting incomplete solutions.
5. Copy some exercise information from [JavaScript `config.json` file](https://github.com/exercism/javascript/blob/master/config.json) to the [ECMAScript `config.json` file](https://github.com/exercism/ecmascript/blob/master/config.json). This information is an object in the `exercises` array property with a `slug` property equals to `run-length-encoding`. Copy that object and place it in the same (or most similar) position in the ECMAScript `config.json` file. Both `config.json` files are in the root folder of the repo.

@exercism/ecmascripts - Do you miss something here?

**Comment by matthewmorgan** [comment]
@rchavarria as I mentioned in the JS track, we also need to change the test suite methods to use Jest here instead of Jasmine, so `it` becomes `test` and `xit` becomes `xtest`.

Thanks for doing this!

**Comment by skabbass1** [comment]
Would like to help with this. Please assign to me. Thanks

**Comment by joelwallis** [comment]
Hi @skabbass1!

There's no need to assign it to you, it's already yours! 😄 Just fork the project, create a new branch and start working on a solution. Open a pull request referencing this issue when you have something ready. We'll love to receive your contribution! 😃

**Comment by matthewmorgan** [comment]
@skabbass1 the one thing you should do is open a PR with an empty commit.  Give it a message 'Dibs - I will add exercise run-length-encoding'.  That will make the fact that you're using it visible to other contributors.  Good luck!  let us know if you need any help.

**Comment by joelwallis** [comment]
@skabbass1 to make an empty commit, you use the `--allow-empty` flag for `git-commit`:

```
$ git commit --allow-empty 'Adds run-length-encoding exercise'
```

Then, go ahead and port the exercise. Remember to fork the project and create a new branch to work on. You can add things to this commit by using the `--amend` flag:

```
$ git commit --amend
```

_PS: the commit message below is just a suggestion._

**Comment by skabbass1** [comment]
Thanks @joelwallis  and @matthewmorgan !

**Comment by skabbass1** [comment]
@joelwallis @matthewmorgan 
In the  *.spec.js files for each of the assignments,  I see that the typical pattern is to import the assignment package name. For example, for the accumulate assignment:
```javascript
import accumulate from './accumulate';

```
The  issue that I am running into with this is that I cannot run the tests from within the assignment directory.  Jest is unable to  resolve the imports:

```javascript
// in ecmascript/exercises/accumulate

npm test
> jest --no-cache ./*

 FAIL  ./accumulate.spec.js
  ● Test suite failed to run

    Cannot find module './accumulate' from 'accumulate.spec.js'

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:179:17)
      at Object.<anonymous> (accumulate.spec.js:1:121)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        0.867s
Ran all test suites matching /./README.md|./accumulate.spec.js|./example.js|./node_modules|./package-lock.json|./package.json/.
npm ERR! Test failed.  See above for more details.

```
If I change the import in the *.spec.js file to:
```javascript
import accumulate from './example'
```
Then the ```npm test``` in the assignment directory works as expected.

Running tests via the ```make test-assignment ASSIGNMENT=accumulate``` works  without any issues though.

Am I missing a step or doing something  incorrectly when running tests in the assignment directory?

**Comment by skabbass1** [comment]
Ah! I see that the make  command renames the example.js file to the corresponding [assignment].js file when it copies files  for testing in in the tmp directory exercism/ecmascript/exercises/[assignment]/tmp. Thats why  ```make test-assignment ASSIGNMENT=[assignment]``` works.

**Comment by skabbass1** [comment]
@joelwallis @matthewmorgan 
I updated the the initial pull request with the changes required for this issue. Take a look and let me know if it is line with what you expect or if you need me to make any changes. Thanks

**Comment by matthewmorgan** [comment]
@skabbass1 I'll have a look at your PR.  Thanks!

BTW on the testing, It sounds like you're clear, but I just want to make sure.  I also am wondering if the contributor docs could be more lucid:
1. `npm test` is for the _users_ to test _their_ solution locally.
2. `make test-assignment` is for _contributors_ to test their contributed test suite and example exercise.

Let me know if there's something you feel we could do to make this less confusing.

**Comment by skabbass1** [comment]
@matthewmorgan 
I think making the contributor docs more  explicit per your suggestion above would be great.

**Comment by apapirovski** [comment]
Same as proverb, looks like this is done.

**Comment by rchavarria** [comment]
Yep, it's true. Thanks for helping on keeping issue list clean 👍 

Closed by #335
